packages/core/src/sync/sync-api.ts

```diff
@@ -77,7 +82,7 @@ export interface SyncStatus {
 
 export class SyncApi implements ISyncApi {
   public readonly modelsToSync = new Set<string>()
-
+  public readonly modelsToHistoricSync = new Map<string, number>()
```

Comment:
This needs to be incremented and decremented for every history sync job that prevents queries.  Right now its incremented only during `startModelSync` and decremented for ALL `history sync` (catch up, reorg, first time, and new model. This will result in the counts being messed up.